acceptance/tests/vault/vault_test.go

```diff
@@ -111,6 +114,20 @@ func TestVault(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, caConfig.Provider, "vault")
 
+	// Validate that tls.crt is not set to the default value
+	logger.Log(t, "Validating that CONSUL_CACERT has been set correctly")
+	serverPod, err := ctx.KubernetesClient(t).CoreV1().Pods(ns).Get(context.Background(), fmt.Sprintf("%s-consul-server-0", consulReleaseName), metav1.GetOptions{})
+	require.NoError(t, err)
+
+	// Environment variables are buried in the pod spec
+	containerVars := serverPod.Spec.Containers[0].Env
+	for k := range containerVars {
+		if containerVars[k].Name == "CONSUL_CACERT" {
+			require.NotEqual(t, containerVars[k].Value, "/consul/tls/ca/tls.crt")
+			require.Equal(t, containerVars[k].Value, "/vault/secrets/serverca.crt")
```

Comment:
❤️

Reply:
I wonder if the assertion should be more behavioral. This feels similar to what we already do in bats where we check that the rendered template has the right env vars.

What if we change it to exercise a user workflow instead? In this case, it'll be:
```
When I exec into consul server container, I should be able to run consul CLI commands and not see errors.
```

Note: We will also need to provide an ACL token to the consul CLI command if we were to do that.

Reply:
Behavioural would probably catch a lot more as it would ensure that everything works...